File: src/test/java/name/seguri/java/leetcode/_20/SolutionTest.java
package name.seguri.java.leetcode._20;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class SolutionTest {

  private final Solution sut = new Solution();

  @Test
  void input1() {
    var input = "()";

    var result = sut.isValid(input);

    assertThat(result).isTrue();
  }

  @Test
  void input2() {
    var input = "()[]{}";

    var result = sut.isValid(input);

    assertThat(result).isTrue();
  }

  @Test
  void input3() {
    var input = "(]";

    var result = sut.isValid(input);

    assertThat(result).isFalse();
  }

  @Test
  void input4() {
    var input = "([])";

    var result = sut.isValid(input);

    assertThat(result).isTrue();
  }

  @Test
  void input5() {
    var input = "[";

    var result = sut.isValid(input);

    assertThat(result).isFalse();
  }
}

File: src/test/kotlin/leetcode/_20/SolutionTest.kt
package leetcode._20

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class SolutionTest {

  private val sut = Solution()

  @Test
  fun input1() {
    val input = "()"

    val result = sut.isValid(input)

    assertThat(result).isTrue()
  }

  @Test
  fun input2() {
    val input = "()[]{}"

    val result = sut.isValid(input)

    assertThat(result).isTrue()
  }

  @Test
  fun input3() {
    val input = "(]"

    val result = sut.isValid(input)

    assertThat(result).isFalse()
  }

  @Test
  fun input4() {
    val input = "([])"

    val result = sut.isValid(input)

    assertThat(result).isTrue()
  }

  @Test
  fun input5() {
    val input = "["

    val result = sut.isValid(input)

    assertThat(result).isFalse()
  }
}
